packages/cli/src/build/utils.ts

```diff
@@ -1,7 +1,40 @@
+import { Project } from "../project";
+import { isTsPath } from "../rollup-plugins/typescript-declarations";
+import { writeDevTSFile } from "../dev";
+import * as fs from "fs-extra";
+import path from "path";
+
 export function getDevPath(cjsPath: string) {
   return cjsPath.replace(/\.js$/, ".dev.js");
 }
 
 export function getProdPath(cjsPath: string) {
   return cjsPath.replace(/\.js$/, ".prod.js");
 }
+
+export async function cleanProjectBeforeBuild(project: Project) {
+  await Promise.all(
+    project.packages.map(async (pkg) => {
+      await Promise.all([
+        fs.remove(path.join(pkg.directory, "dist")),
+        ...pkg.entrypoints
+          .filter((entrypoint) => entrypoint.name !== pkg.name)
+          .map((entrypoint) => {
+            return fs.remove(path.join(entrypoint.directory, "dist"));
+          }),
+      ]);
+
+      await Promise.all(
+        pkg.entrypoints.map(async (entrypoint) => {
+          if (isTsPath(entrypoint.source)) {
+            await fs.mkdir(path.join(entrypoint.directory, "dist"));
+            await writeDevTSFile(
+              entrypoint,
+              await fs.readFile(entrypoint.source, "utf8")
+            );
+          }
+        })
+      );
```

Comment:
is this required for the watch script? can it result in some unwanted side-effects? why this is needed for the build script?

Reply:
It's needed because we build everything in parallel so when doing a build, the TypeScript compiler needs to be able to resolve packages before they're built.

Reply:
I guess this doesn't create problems that much so it's OK. Out of curiosity - have you considered using `paths` options in the tsconfig (the one created/loaded programmatically and passed to the TS API)? Or have you not wanted to increase the complexity there?

Reply:
I tried but couldn't get it to work with paths ¯\\\_(ツ)_/¯